src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala

```diff
@@ -617,22 +619,113 @@ object Helpers extends SparkSessionWrapper {
    *                        or direct access via s3:// or abfss:// or dbfs:/mnt/ etc.
    * @return
    */
+//  def registerRemoteOverwatchIntoLocalMetastore(
+//                                                 remoteWorkspace: Workspace,
+//                                                 localDataTarget: DataTarget
+//                                               ): Seq[WorkspaceMetastoreRegistrationReport] = {
+//
+//    val newConfigParams = remoteWorkspace.getConfig.inputConfig.copy(dataTarget = Some(localDataTarget))
+//    val newConfigArgs = JsonUtils.objToJson(newConfigParams).compactString
+//    val localTempWorkspace = Initializer(newConfigArgs, disableValidations = true)
+//    val registrationReport = localTempWorkspace.addToMetastore()
+//    val b = Bronze(localTempWorkspace, suppressReport = true, suppressStaticDatasets = true)
+//    val g = Gold(localTempWorkspace, suppressReport = true, suppressStaticDatasets = true)
+//
+//    b.refreshViews()
+//    g.refreshViews()
+//    registrationReport
+//
+//  }
   def registerRemoteOverwatchIntoLocalMetastore(
-                                                 remoteWorkspace: Workspace,
-                                                 localDataTarget: DataTarget
-                                               ): Seq[WorkspaceMetastoreRegistrationReport] = {
+                                                 remoteStoragePrefix: String,
+                                                 remoteWorkspaceID: String,
+                                                 localETLDatabaseName: String = "",
+                                                 localConsumerDatabaseName: String = "",
+                                                 remoteETLDataPathPrefixOverride: String = "",
+                                                 usingExternalMetastore: Boolean = false,
+                                                 workspacesAllowed: Array[String] = Array()
+                                               ): Seq[WorkspaceMetastoreRegistrationReport]  = {
+
+    // Derive eltDatapathPrefix
+    val eltDataPathPrefix = if (remoteETLDataPathPrefixOverride == "") {
+      remoteStoragePrefix + "/global_share"
+    } else remoteETLDataPathPrefixOverride
+
+    // Check whether eltDatapathPrefix Contains PipReport
+    val pipReportPath = eltDataPathPrefix+"/pipeline_report"
+    try{
+      dbutils.fs.ls(pipReportPath)
+      logger.log(Level.INFO, s"${pipReportPath} is a Delta table.....Proceed")
+    }catch {
+
+      case e: FileNotFoundException =>
+        val msg = s"${pipReportPath} is not a Delta table.....Exit"
+        logger.log(Level.ERROR, msg)
+        throw new BadConfigException(msg)
+    }
+
+    // Derive Remote Workspace
+    val remoteWorkspace = Helpers.getRemoteWorkspaceByPath(pipReportPath,remoteWorkspaceID)
+
+    val remoteConfig = remoteWorkspace.getConfig
```

Comment:
will review this deeper section later after first round. Want to do testing but not ready quite yet.